CAN_Decoder_SourceCode_TestRel_21Dec21.py

```diff
@@ -0,0 +1,107 @@
+import tkinter as tk
+from tkinter import *
+from tkinter import filedialog
+from datetime import datetime
+
+
+# Function for opening the file
+def CAN_Decoder():
+    lineNum = 0
+    oldDataFile = filedialog.askopenfile(initialdir="/")
+    listedOldData = []
+    for dataLine in oldDataFile:
+        segmentedDataLine = dataLine.split(" ")
+        listedOldDataLine = []
+        for stringSegment in segmentedDataLine:
+            if stringSegment.strip():
+                listedOldDataLine.append(stringSegment)
+        listedOldData.append(listedOldDataLine)
+        lineNum = lineNum + 1
+
+    newData = []
```

Comment:
A good idea will be to separete the "input data parsing" from "TP frame reassembly" into 2 separate functions.

Reply:
Implemented. Please refer V1.1